### PR TITLE
ci: add gosec to PRs and main

### PR DESCRIPTION
**Pull request by prajjwol**
Closes: ENG-255

## Description

Adding `security.yml` in `.github/workflows` to enable [Gosec](https://linear.app/tharsis/issue/ENG-255/add-gosec-github-action-to-check-security-issues-on-sdk-based-chains)

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by prajjwol** [comment]
We have 16 issues in main at the moment. Need to fix (or ignore) them before this goes in.

**Comment by fedekunze** [comment]
cc: @odeke-em

File: .github/workflows/security.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          args: ./...
+          # we let the report trigger content trigger a failure using the GitHub Security features.
+      #     args: '-no-fail -fmt sarif -out results.sarif ./...'
+      # - name: Upload SARIF file
+      #   uses: github/codeql-action/upload-sarif@v1
+      #   with:
+      #     # Path to SARIF file relative to the root of the repository
+      #     sarif_file: results.sarif
+        # if: "env.GIT_DIFF != ''"
```
Comment (fedekunze) [inline .github/workflows/security.yml:34]: @prajjwol follow up we should enable these comments

File: .github/workflows/security.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Checkout Source
+        uses: actions/checkout@v2.4.0
+      # - uses: technote-space/get-diff-action@v5
+      #   with:
+      #     SUFFIX_FILTER: |
+      #       .go
+      #       .mod
+      #       .sum
+      - name: Run Gosec Security Scanner
+        uses: informalsystems/gosec@master
```
Comment (fedekunze) [inline .github/workflows/security.yml:24]: @odeke-em I assume we need to use the informal version? do we need to also add a custom rule for the SDK?

**Review by fedekunze — APPROVED** [review]
ACK. Let's add the follow-up changes and fix the gosec issues

**Pull request merged**